### PR TITLE
gfortran needs -lpthread & -maix64(64 build) in AIX

### PR DESCRIPTION
**Pull request by ayappanec**
Please see #12396

### CODE REVIEW

**Comment by mattip** [comment]
@ayappanec We do not have a AIX system in our CI matrix, so the new lines are not covered by any tests. I assume you or someone with an AIX system checked that this works?

**Comment by ayappanec** [comment]
I am from AIX Opensource development team (IBM). We are the team driving IBM AIX Toolbox https://www.ibm.com/developerworks/aix/library/aix-toolbox/alpha.html
I have tested this very well in AIX.

**Comment by tylerjereddy** [comment]
@ayappanec I'm guessing there are currently no easy ways to include AIX in our test / CI matrix? Like a docker container that runs AIX on x86-64 or something like that?

File: numpy/distutils/fcompiler/gnu.py [hunk 1]
```diff
@@ -315,6 +315,12 @@ def version_match(self, version_string):
     module_dir_switch = '-J'
     module_include_switch = '-I'
 
+    if sys.platform[:3] == 'aix':
+        executables['linker_so'].append('-lpthread')
```
Comment (rgommers) [inline numpy/distutils/fcompiler/gnu.py]: Is this needed for all Fortran code, or only for (e.g.) BLAS?

The only other place we have this is for MKL BLAS (defined in `distutils/system_info.py`).

File: numpy/distutils/fcompiler/gnu.py [hunk 2]
```diff
@@ -315,6 +315,12 @@ def version_match(self, version_string):
     module_dir_switch = '-J'
     module_include_switch = '-I'
 
+    if sys.platform[:3] == 'aix':
```
Comment (rgommers) [inline numpy/distutils/fcompiler/gnu.py]: If this stays in this file (referring to my question on BLAS), this is okay - although by design the better place is probably to add the method `get_flags_linker_so` et al. - but given the number of sets of flags to be edited, that's not necessarily more understandable this way.

File: numpy/distutils/fcompiler/gnu.py (hunk 1 above)
Comment (ayappanec) [reply numpy/distutils/fcompiler/gnu.py]: Yes, this is needed for Fortran code. I added little more info about this here --> https://github.com/numpy/numpy/issues/12396#issuecomment-439348349

File: numpy/distutils/fcompiler/gnu.py (hunk 2 above)
Comment (ayappanec) [reply numpy/distutils/fcompiler/gnu.py]: Okay. I want to make the additions in one place rather than in various places.

**Comment by rgommers** [comment]
Great, in it goes. Thanks @ayappanec

**Comment by ayappanec** [comment]
Thank You @rgommers

**Comment by ayappanec** [comment]
@tylerjereddy AIX runs only on IBM Power systems. IBM has this Power development cloud platform ( https://www-356.ibm.com/partnerworld/wps/servlet/ContentHandler/stg_com_sys_power-development-platform ) where one can register and get AIX machines at no charge.

**Pull request merged**